game-core/src/main/java/games/strategy/engine/data/changefactory/units/UnitHitsChange.java

```diff
@@ -0,0 +1,57 @@
+package games.strategy.engine.data.changefactory.units;
+
+import games.strategy.engine.data.Change;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import org.triplea.java.collections.IntegerMap;
+
+/** A game data change that captures the damage done to a collection of units. */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class UnitHitsChange extends Change {
+  private static final long serialVersionUID = 2862726651812142713L;
+
+  private final IntegerMap<String> newHits;
+  private final IntegerMap<String> oldHits;
```

Comment:
IntegerMap is more than just a Map<T, Integer>.  It allows easy adding and summing of the integers.  So it shouldn't be deprecated.

But for this specific case, since it is just storing the IntegerMap data, it can be turned into a Map.

Reply:
The summing operations are not necessarily that difficult. A util class could replace the functionality.

The 'IntegerMap' is kinda counter-intuitive, I've generally found it takes a good bit of extra thought to determine what it represents. I think the deprecation is on the table, but maybe not replacing with `map` but a first class data structure. A `map` and `IntegerMap` seems to be very strutc-like, weak types and very low level.